### ISSUE TITLE
Warning or error before saving a dataset to a directory with pre-existing files

### ISSUE THREAD

**Issue by puddleglum56** [deepchem/deepchem#3402]
## 🚀 Feature

It would be nice to have some kind of warning or error before a dataset is saved to a directory if the directory already contains files. An overwrite=True option could remove the error or warning.

## Motivation

As a user, it was quite shocking to innocently `move` a numpy dataset to a directory and `save_to_disk` only to find the directory completely overwritten and replaced with my deepchem dataset. Luckily most of my directory was backed up. Hopefully this shock could be avoided (especially for new users) by adding some sort of warning or error before overwriting the directory.

## Additional context

N/A

**Comment by rbharath** [comment]
I'm sorry to hear about the issue here! I agree this could definitely be improved with a warning/error if the directory is not empty. I've marked this as open for contribution

**Comment by datamit-tech** [comment]
@rbharath I can take this up, kindly assign this to me

**Comment by ioannastantzou** [comment]
Hello! I would like to ask whether this issue is eventually fixed or if I could be assgined for it.

**Comment by rbharath** [comment]
@ioannastantzou Can you come by office hours? I can discuss with you there

**Comment by Plomo-02** [comment]
Hello, is this issue still open? Can I be assigned for it? i'de like to add this "feature"

**Comment by Anuja0412Gupta** [comment]
Hi! I would like to ask whether this issue is  fixed or if I could be assgined for it.

**Comment by Jadhavsakshi7887** [comment]
i am new in open source can you tell me in which file that issue exists i am unable to find

**Comment by sirine707** [comment]
Hi @Jadhavsakshi7887 are you currently working on this issue ?

**Comment by shostykovich21** [comment]
Hi @rbharath. I can work on this issue. If the issue's still available, can you please assign this to me.
Thanks.

**Comment by NitinRwt** [comment]
Hi @rbharath. I can work on this issue, can you please assign this to me.
thansks .

**Comment by prasanna7codes** [comment]
you can ignore this , I am just typing random things so I can learn how these comments and other things function

**Comment by Vinay989724** [comment]
please assign this issue to me ...

**Comment by Jadhavsakshi7887** [comment]
No

**Comment by prasanna7codes** [comment]
@rbharath [this is my PR link ](https://github.com/deepchem/deepchem/pull/4256), I have done the changes and have also added test cases

**Comment by JanaviN7** [comment]
Hi, I see a PR was submitted for this issue. Is there any additional work or improvements needed? I’d be happy to contribute.

**Comment by Aditya66050177** [comment]
@rbharath Hey if the issue is still open ,i would like to make improvements for this

**Comment by likhithapjb** [comment]
dear mentor,
   
i hope you are doing well.i am highly interested in contributing to "rao lab" as part of google summer of code 2025.
i am particicularly enthusiastic about warning or error before saving a dataset to a directory with pre-existing files and would love to learn more about how i can get started.


i would greatly appreciate any guidence or suggestions on how i can contribute effectively .looking forward to your response.


best regards,
likhitha narayana

**Comment by nishaant02** [comment]
Hiii @rbharath ,
Hope you are doing well . i can contribute to this issue if you allow me 

1. We will check if the directory already  exists and contains files before saving.

2. We will raise a warning or error when a non-empty directory is detected.

3. We will introduce an '**overwrite=True**' option that allows bypassing the warning if the user explicitly wants to overwrite.

By this approach , we can start solving the issue and i can provide a pseudocode also for this .
There are many projects in which i can contribute which you registered in GSOC'25 . 
I would love to work under your guidance and contribute .
Looking forward to your positive approach .

Best Regards ,
Nishant Gupta

**Comment by DAKSH-coder26** [comment]
Hi! I'd like to work on this.

My plan is:
- Detect if the target directory exists and is non-empty
- Raise a warning by default (or error if maintainers prefer)
- Add an `overwrite=True` option to skip the warning

This keeps current behavior intact while preventing accidental data loss.  
Please let me know if there's anything specific I should watch out for.

**Comment by KariraLakshya** [comment]
hey @rbharath , can i work on this issue if this is still open??

**Comment by arjitrawat15** [comment]
Hey @rbharath 

This looks like a very useful safety improvement, especially for new users. I’d like to work on this issue.
My initial thought is to add a check in save_to_disk that detects whether the target directory already exists and contains files, and then either:

1. raise a clear error / warning by default, or
2. allow overwriting only when an explicit overwrite=True flag is provided.
I’ll make sure the change is backward-compatible and add tests to cover the new behavior.
Please let me know if this approach aligns with the maintainers’ expectations, or if there’s a preferred design for handling this case.

**Comment by khushi-070906** [comment]
Dear DeepChem Maintainers,

Interested in contributing to DeepChem as a beginner contributor

Technical skills:
Python
PyTorch
Machine Learning & Deep Learning fundamentals
Matplotlib (data visualization), Numpy,Pandas, Tensorflow,Scikit-Learn
Seaborne, OpenCV 
LLMs and Retrieval-Augmented Generation (RAG)
Computer vision–based ML projects
Built applied AI projects (AI legal assistant, automated attendance system)

Currently exploring the DeepChem repository

Seeking beginner-friendly / good first issues and guidance on where to contribute

Thank you for your time.

Kind regards,
Khushi Mittal
Email: khushimittal070906@gmail.com
